src/Core/StaticConfigTrait.php

```diff
@@ -249,18 +249,23 @@ public static function parseDsn($dsn)
             throw new InvalidArgumentException('Only strings can be passed to parseDsn');
         }
 
-        $scheme = '';
-        if (preg_match("/^([\w\\\]+)/", $dsn, $matches)) {
-            $scheme = $matches[1];
-            $dsn = preg_replace("/^([\w\\\]+)/", 'file', $dsn);
-        }
+        $pattern = '/^(?P<scheme>[\w\\\\]+):\/\/((?P<user>.*?)(:(?P<password>.*?))?@)?' .
+            '((?P<host>[.\w\\\\]+)(:(?P<port>\d+))?)?' .
+            '(?P<path>\/[^?]*)?(\?(?P<query>.*))?$/';
+        preg_match($pattern, $dsn, $parsed);
 
-        $parsed = parse_url($dsn);
-        if ($parsed === false) {
-            return $dsn;
+        if (empty($parsed)) {
+            return false;
```

Comment:
What's the use case for returning a boolean here? The return line in the doc-block specifies only array. Since this is ?only? used for configuration wouldn't you expect some kind of error if the DSN you've passed can't be parsed?